src/main/java/com/echobox/api/linkedin/client/DefaultWebRequestor.java

```diff
@@ -168,6 +182,11 @@ public DefaultWebRequestor(String clientId, String clientSecret) throws GeneralS
   public DefaultWebRequestor(String clientId, String clientSecret, String accessToken)
       throws GeneralSecurityException, IOException {
     this.requestFactory = authorize(clientId, clientSecret, accessToken);
+    this.headers = accessToken != null ? Arrays.asList("Authorization",
+        String.format("Bearer %s", accessToken)) : Collections.emptyList();
+    this.httpClient = HttpClient.newBuilder().followRedirects(HttpClient.Redirect.NORMAL)
+        .connectTimeout(Duration.ofMillis(DEFAULT_CONNECT_TIMEOUT_MS)).build();
+    this.readTimeout = DEFAULT_READ_TIMEOUT_MS;
```

Comment:
Constructor will change to include connect and read timeouts as parameters in follow-up PR; didn't want to do too much at once.